content/guides/kubernetes/app-enhancements-probing-app-state.md

```diff
@@ -0,0 +1,199 @@
+configuration is as follows.
+
+```yaml
+readinessProbe:
+  httpGet:
+    path: /ready
+    port: 8080
+  initialDelaySeconds: 5
+  periodSeconds: 10
+  failureThreshold: 2
+```
+
+The following diagram demonstrates the impact of this `readinessProbe`.
+
+![readiness probe](/images/guides/kubernetes/app-enhancements/readiness-probe.png)
+
+## Determining Application Liveness
+
+Liveness represents whether the application is healthy. If it is determined unhealthy, the 
+container failing the check is restarted. As applications run for a long duration, it is essential
+these checks are in place.
+
+In Kubernetes, readiness probes are configured in `spec.containers[i].livenessProbe`. They are 
+nearly identical to `readinessProbe`s, but upon failure restart the container rather than just
+halting traffic. An example configuration is as follows.
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /healthz
+    port: 8080
+  initialDelaySeconds: 60
+  periodSeconds: 10
+  failureThreshold: 10
+```
+
+The following diagram demonstrates the impact of this `livenessProbe`.
+
+![liveness-probe](/images/guides/kubernetes/app-enhancements/liveness-probe.png)
+
+### Startup Probes - Protecting Slow Container Initialization
+> Startup Probes are not enabled by default until Kubernetes 1.18. 
+Until then, they need to be enabled via `--feature-gates` on the 
+`kube-apiserver` and the `kubelet` in each node.
+
+Sometimes a `livenessProbe` might be ideal as an ongoing check, 
+but not necessarily for initialization time. 
+
+Some applications might require additional time during startup, 
+for example when warming up a cache or preparing their 
+data store for the first time. In these cases, the 
+`livenessProbe`'s `failureThreshold` and `periodSeconds` settings 
+might turn out to be too short, and increasing them could 
+diminish the value of the checks through the container's lifetime.
+
+The need for an initialization-only check is covered with 
+a `startupProbe`.  It can be configured with the same or a different
+command, TCP, or HTTP check as your `livenessProbe` and with longer 
+thresholds, long enough to cover the worst-case 
+scenario startup time.
+ 
+Our previous example could be modified to look as follows.
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /healthz
+    port: 8080
+  periodSeconds: 10
+  failureThreshold: 10
+
+startupProbe:
+  httpGet:
+    path: /cache/healthz
+    port: 8080
+  periodSeconds: 10
+  failureThreshold: 30
+```
+
+Note that `initialDelaySeconds` can be removed from the 
+`livenessProbe` as we're using `startupProbe` to cover our
+Pod initialization checks. In the example above, the `startupProbe`
+is allowing the container 300 seconds (`periodSeconds` * `failureThreshold`)
+to finish its startup. It is also validating against a different
+endpoint than the `livenessProbe`.
+
+![startup probe](/images/guides/kubernetes/app-enhancements/startup-probe.png)
```

Comment:
For images with transparent backgrounds (or even ones that need some padding) you can add them to a subfolder under `diagrams` and explained in the README (https://github.com/vmware-tanzu/tanzu-dev-portal#images). This makes it work okay in both light and dark mode.